apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-action.tsx

```diff
@@ -0,0 +1,198 @@
+import { ComponentProps } from 'react';
+import { useFormContext, useWatch } from 'react-hook-form';
+import { RiEdit2Line, RiExpandUpDownLine, RiForbid2Line } from 'react-icons/ri';
+import { liquid } from '@codemirror/lang-liquid';
+import { EditorView } from '@uiw/react-codemirror';
+
+import { Button, buttonVariants } from '@/components/primitives/button';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '@/components/primitives/dropdown-menu';
+import {
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+  FormMessagePure,
+} from '@/components/primitives/form/form';
+import { InputField } from '@/components/primitives/input';
+import { Popover, PopoverContent, PopoverTrigger } from '@/components/primitives/popover';
+import { Separator } from '@/components/primitives/separator';
+import { URLInput } from '@/components/workflow-editor/url-input';
+import { cn } from '@/utils/ui';
+import { urlTargetTypes } from '@/utils/url';
+import { Editor } from '@/components/primitives/editor';
+
+const primaryActionKey = 'primaryAction';
+const secondaryActionKey = 'secondaryAction';
+
+export const InAppAction = () => {
```

Comment:
This file and component were renamed from `ActionPicker`.
The changes that were added are: the form fields and errors; handling onChange logic;